src/fsharp/MethodOverrides.fs

```diff
@@ -255,23 +255,40 @@ module DispatchSlotChecking =
 
         let isReqdTyInterface = isInterfaceTy g reqdTy 
         let showMissingMethodsAndRaiseErrors = (isReqdTyInterface || not isOverallTyAbstract)
-        let res = ref true
-        let fail exn = (res := false ; if showMissingMethodsAndRaiseErrors then errorR exn)
         
+        let mutable res = true
+        let fail exn =
+            res <- false
+            if showMissingMethodsAndRaiseErrors then
+                errorR exn
+
+        let rec listSameLength l1 l2 = 
```

Comment:
Just use `List.length l1 = List.length l2` or `l1.Length = l2.Length`. No need for a new function